### PR TITLE
Port pywatchman to Python 3

### PR DESCRIPTION
**Pull request by sid0**
This branch will contain work to port pywatchman to Python 3. It's currently a work in progress.

### CODE REVIEW

**Comment by facebook-github-bot-4** [comment]
By analyzing the blame information on this pull request, we identified @wez, @bhamiltoncx and @mjpieters to be potential reviewers.

**Comment by facebook-github-bot-8** [comment]
@sid0 updated the pull request.

**Comment by facebook-github-bot-7** [comment]
@sid0 updated the pull request.

**Comment by facebook-github-bot-3** [comment]
@sid0 updated the pull request.

**Comment by facebook-github-bot-8** [comment]
@sid0 updated the pull request.

**Comment by facebook-github-bot-9** [comment]
@sid0 updated the pull request.

**Comment by facebook-github-bot-5** [comment]
@sid0 updated the pull request.

**Comment by facebook-github-bot-0** [comment]
@sid0 updated the pull request.

File: python/pywatchman/bser.c [hunk 1]
```diff
@@ -910,14 +918,14 @@ static PyObject *bser_loads(PyObject *self, PyObject *args)
   int datalen = 0;
   const char *end;
   int64_t expected_len;
-  int mutable = 1;
   PyObject *mutable_obj = NULL;
+  unser_ctx_t ctx = {1};
 
-  if (!PyArg_ParseTuple(args, "s#|O:loads", &data, &datalen, &mutable_obj)) {
+  if (!PyArg_ParseTuple(args, "s#|Os:loads", &data, &datalen, &mutable_obj)) {
```
Comment (wez) [inline python/pywatchman/bser.c]: Was this intentional?  I don't see `Os` in my python 2 docs

File: python/tests/tests.py [hunk 2]
```diff
@@ -40,16 +41,13 @@ def id(self):
 
     def roundtrip(self, val, mutable=True):
         enc = self.bser_mod.dumps(val)
-        print "# %s  -->  %s" % (val, enc.encode('hex'))
+        print "# %s  -->  %s" % (repr(val), enc.encode('hex'))
```
Comment (wez) [inline python/tests/tests.py]: does this mean that we need to implement `__repr__` in our C bser tuple stuff?

**Comment by facebook-github-bot-7** [comment]
@sid0 updated the pull request.

File: python/tests/tests.py (hunk 2 above)
Comment (sid0) [inline python/tests/tests.py]: There is a default `repr` implementation, but in any case `val` should never be a bser tuple.

File: python/tests/tests.py [hunk 3]
```diff
@@ -6,6 +6,8 @@
 import os
 from pywatchman import bser, pybser, SocketTimeout, WatchmanError
 
+PILE_OF_POO = u"\U0001F4A9"
```
Comment (wez) [inline python/tests/tests.py]: strong hire

Comment (mjpieters) [inline python/tests/tests.py]: Pfff, I would have been more impressed had he used used `u'\U0001F596'` instead :wink:

**Comment by facebook-github-bot-2** [comment]
@sid0 updated the pull request.

**Comment by wez** [comment]
lgtm

**Pull request merged**